plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt

```diff
@@ -1017,8 +782,166 @@ class GesturePluginTest {
+    every { mapDelegateProvider.mapTransformDelegate } returns mapTransformDelegate
+    every { mapDelegateProvider.mapPluginProviderDelegate } returns mapPluginProviderDelegate
+    every { mapPluginProviderDelegate.getPlugin<CameraAnimationsPlugin>(Plugin.MAPBOX_CAMERA_PLUGIN_ID) } returns cameraAnimationsPlugin
+
+    presenter = GesturesPluginImpl(context, attrs, mockk(relaxed = true))
+
+    presenter.bind(context, gesturesManager, attrs, 1f)
+    presenter.onDelegateProvider(mapDelegateProvider)
+    presenter.initialize()
+
+    every { mapCameraManagerDelegate.cameraState } returns CameraState(
+      Point.fromLngLat(0.0, 0.0),
+      EdgeInsets(0.0, 0.0, 0.0, 0.0),
+      0.0,
+      0.0,
+      targetPitch
+    )
+    every {
+      mapCameraManagerDelegate.getDragCameraOptions(
+        any(),
+        any()
+      )
+    } returns CameraOptions.Builder().build()
+    presenter.updateSettings { scrollMode = targetScrollMode }
+    every { motionEvent.x } returns 0.0f
+    every { motionEvent.y } returns 0.0f
+  }
+
+  @Test
+  fun testFling() {
+
+    val result = presenter.handleFlingEvent(motionEvent, mockk(), velocity.first, velocity.second)
+    verify {
+      mapCameraManagerDelegate.getDragCameraOptions(
+        ScreenCoordinate(0.0, 0.0),
+        expectedCoordinate
+      )
+    }
+    verify {
+      cameraAnimationsPlugin.easeTo(
+        CameraOptions.Builder().build(),
+        capture(mapAnimationOptionsSlot)
+      )
+    }
+    assert(result)
+    assertEquals(MapAnimationOwnerRegistry.GESTURES, mapAnimationOptionsSlot.captured.owner)
+    assertEquals(expectedFlingDuration, mapAnimationOptionsSlot.captured.duration)
+  }
 
   private companion object {
+    @JvmStatic
+    @ParameterizedRobolectricTestRunner.Parameters(name = "Fling at pitch {0} with scroll mode {1} and velocity {2} should end at screen coordinate {3} and fling duration should be {4}")
+    fun data() = listOf(
+      arrayOf(
+        0.0,
+        ScrollMode.HORIZONTAL_AND_VERTICAL,
+        Pair(FLING_VELOCITY, FLING_VELOCITY),
+        ScreenCoordinate(FLING_DISPLACEMENT, FLING_DISPLACEMENT),
+        1414L
+      ),
+      arrayOf(
+        55.0,
+        ScrollMode.HORIZONTAL_AND_VERTICAL,
+        Pair(FLING_VELOCITY, FLING_VELOCITY),
+        ScreenCoordinate(645.1612903225806, 645.1612903225806),
+        912L
+      ),
+      arrayOf(
+        80.0,
+        ScrollMode.HORIZONTAL_AND_VERTICAL,
+        Pair(FLING_VELOCITY, FLING_VELOCITY),
+        ScreenCoordinate(67.93869850933991, 67.93869850933991),
+        96L
+      ),
+      arrayOf(
+        0.0,
+        ScrollMode.HORIZONTAL_AND_VERTICAL,
+        Pair(800f, 750f),
+        ScreenCoordinate(80.0, 75.0),
+        109L
+      ),
+      arrayOf(
+        55.0,
+        ScrollMode.HORIZONTAL_AND_VERTICAL,
+        Pair(800f, 750f),
+        ScreenCoordinate(51.61290322580645, 48.38709677419355),
+        70L
+      ),
+      arrayOf(
+        80.0,
```

Comment:
do we need a test for pitch higher than max pitch? or is it impossible?

Reply:
With current gestures implementation, it's impossible to set pitch over 85 using gestures.
And I also double checked if user could set the camera pitch to 85.0 programmatically, and seems the gl-native clamped to the pitch to a valid range(maximum 85.0) as well. So I think it's safe here.